### PR TITLE
fix: snapchat conversion introduce missing fields

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> Resolves INT-794

Some fields were missing from mapping in `snapchat_conversion`. So all the direct general mappings found are included in this PR.
These fields are:
1. click_id
2. first_name
3. middle_name
4. last_name
5. city
6. state
7. country
8. description
9. customer_status
10. uuid_c1
11. level

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/snapchat_conversion/transform.js [hunk 1]
```diff
@@ -190,11 +209,19 @@ function trackResponseBuilder(message, { Config }, mappedEvent) {
   } else {
     throw new InstrumentationError(`Event ${event} doesn't match with Snapchat Events!`);
   }
-
+  payload.description = get(message, 'properties.description');
+  if (Array.isArray(message.properties?.brands)) {
+    payload.brands = get(message, 'properties.brands');
```
Comment (sanpj2292) [inline src/v0/destinations/snapchat_conversion/transform.js:214]: Can we add a test-case to cover this ?

**Review by sanpj2292 — APPROVED** [review]

File: src/v0/destinations/snapchat_conversion/transform.js (hunk 1 above)
Comment (anantjain45823) [reply src/v0/destinations/snapchat_conversion/transform.js:214]: Done

Comment (anantjain45823) [reply src/v0/destinations/snapchat_conversion/transform.js:214]: Done

**Review by mihir-4116 — DISMISSED** [review]

File: src/v0/destinations/snapchat_conversion/transform.js [hunk 2]
```diff
@@ -85,14 +109,82 @@ function populateHashedValues(payload, message) {
...
-  let payload = {};
-  const event = mappedEvent.trim().replace(/\s+/g, '_');
+  if (eventConversionType === 'MOBILE_APP' && !(appId && snapAppId)) {
+    if (!appId) {
+      throw new ConfigurationError('App Id is required for app events');
+    } else {
+      throw new ConfigurationError('Snap App Id is required for app events');
+    }
+  }
+};
```
Comment (sanpj2292) [inline src/v0/destinations/snapchat_conversion/transform.js:137]: [Minor]
```suggestion
  if (eventConversionType === 'MOBILE_APP') {
    let requiredId = "App Id"
    if (!snapAppId) {
      requiredId = "Snap App Id"
    }
    throw new ConfigurationError(`${requiredId} is required for app events`);
  }
};
```

**Review by sanpj2292 — APPROVED** [review]

**Review by mihir-4116 — APPROVED** [review]

**Pull request merged**